glide-core/src/client/value_conversion.rs

```diff
@@ -379,6 +380,68 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        ExpectedReturnType::GeoSearchReturnType(dist, hash, coord) => match value {
+            Value::Array(array) => {
+                if !dist && !hash && !coord {
+                    return Ok(Value::Array(array));
+                }
+
+                let mut converted_array = Vec::with_capacity(array.len());
+                let num_true = dist as usize + hash as usize + coord as usize;
```

Comment:
num_true doesn't really describe what you're saving here.  This is the expected number of data fields associated with the geo point.  
```suggestion
                let num_rest_data = dist as usize + hash as usize + coord as usize;
```